integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/call/CallMediatorBlockingSenderTestCase.java

```diff
@@ -0,0 +1,246 @@
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("403"), "Expected 403, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 403 Forbidden"),
+                "Required payload not found");
+    }
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking not found http status endpoint with blocking call", enabled = true)
+    public void testNotFoundHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("NotFoundEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("404"), "Expected 404, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 404 Not Found"),
+                "Required payload not found");
+    }
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking conflict http status endpoint with blocking call", enabled = true)
+    public void testConflictHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("ConflictEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("409"), "Expected 409, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 409 Conflict"),
+                "Required payload not found");
+    }
+
+
+    @Test(groups = {"wso2.esb"},
+            description = "Test invoking server error http status endpoint with blocking call", enabled = true)
+    public void testServerErrorHttpStatusEndpoint() throws Exception {
+        URL endpoint = new URL(getProxyServiceURLHttp("ServerErrorEndpointProxy"));
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/plain");
+
+        HttpResponse httpResponse = sendRequest(endpoint, header);
+
+        assertTrue(String.valueOf(httpResponse.getResponseCode()).equals("500"), "Expected 500, found "
+                + httpResponse.getResponseCode() + " http status code");
+        assertTrue(httpResponse.getData().equalsIgnoreCase("Received: 500 Internal Server Error"),
+                "Required payload not found");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        super.cleanup();
+    }
+
+    public static HttpResponse sendRequest(URL endpoint, Map<String, String> headers)
```

Comment:
Shall we move this method to util class and reuse it in all the places?

Reply:
Done in commit [f23167f](https://github.com/wso2/product-ei/pull/3468/commits/f23167fede3bebc4cb3dba75083e82c95eab860c)